.circleci/config.yml

```diff
@@ -35,10 +35,45 @@ commands:
             - run:
                   name: prepare test git user
                   command: git config --global user.email "circleci@example.com" && git config --global user.name "CircleCi Build"
+    run-regressions:
+        parameters:
+            force_update_regression_cache:
+                type: boolean
+                default: false
+            regression_color:
+                type: string
+            regression_scale:
+                type: string
+        steps:
+            - restore_cache:
+                  keys:
+                      - v1-golden-images-<< parameters.regression_color >>-<< parameters.regression_scale >>-b9e157e40b4274683707eccf7f8d400ed4c719bc
+                      - v1-golden-images-master-<< parameters.regression_color >>-<< parameters.regression_scale >>-
+            - run: yarn test:visual:ci --color=<< parameters.regression_color >> --scale=<< parameters.regression_scale >> || << parameters.force_update_regression_cache >>
+            - run: cp -RT test/visual/screenshots-current/ci test/visual/screenshots-baseline/ci
+            - when:
+                  condition: << parameters.force_update_regression_cache >>
+                  steps:
+                      - save_cache:
+                            name: Force Update Regresssion Cache
+                            paths:
+                                - test/visual/screenshots-baseline/ci
+                            key: v1-golden-images-<< parameters.regression_color >>-<< parameters.regression_scale >>-{{ .Revision }}
+            - save_cache:
+                  name: Build Branch Cache
+                  paths:
+                      - test/visual/screenshots-baseline/ci
+                  key: v1-golden-images-{{ .Branch }}-<< parameters.regression_color >>-<< parameters.regression_scale >>-{{ epoch }}
```

Comment:
not sure if theres any additional costs from unnecessary caching, should we look into only making this cache save on master? might not be worth bothering with.

Reply:
To the best of my understanding, we're on the "open source" plan here, which makes CircleCI $free.99, but I'll confirm with the opensource channel.

Reply:
sounds good, even more reason to make sure we're well behaved members of their open source plan!